server/server.js

```diff
@@ -96,113 +100,82 @@ module.exports = function(__dirname) {
 
 	app.post("/editor/:MOD", function(req, res) {
 		if (req.user) {
-			uploadPost(req.params.MOD, req.body, req.user.id).
-				then(renderer.reload).
-				catch(logger.error)
+			uploadPost(req.params.MOD, req.body, req.user.id)
+				.then(renderer.reload.bind(renderer))
+				.catch(logger.error)
 			res.send({msg: "Ok"})
 		} else {
 			req.status(403)
 			res.send("Please Log In first")
 		}
 	})
-	
+
 	/**
 	  * Upload Handling
 	**/
 
-	var uploadPost = function(modify, body, author) {
-		data = {
-			"db": {	
-				"tags": body.tags,
-				"visible": body.visible || false
-			},
-			"content": {
-				"value": body.content
-			}
-		}
-
+	var uploadPost = function(guid, body, author) {
 		if (body.tags.length == 0 && body.title.match(/^\s*$/) && body.content.match(/^\s*$/) && modify) {
-			return new Promise(function(resolve, reject) {
-				db.posts.remove({guid: modify}, function(e) {
-					if (e) {
-						reject(e)
-					} else {
-						logger.log(`Deleting post with id ${modify}`)
-						resolve()
-					}
-				})
-			})
+			return db.posts.remove({ guid });
 		}
 
-		if (!modify) {
-			data.db.author = author
-			data.db.timestamp = Date.now()
-			return utils.generateId(db.posts).then(function (id) {
-				data.db.guid = id
-				data.db.title = {
-					text: body.title,
-					url: `posts/${id}`
-				}
-				data.db.slug = utils.slugify(data.db.title.text)
-				return id
-			}).then(function (id) {
-				return insert(data, false).then( () => id )
-			})
+		if (!guid) {
+			return utils.generateId(db.posts)
+				.then((id) =>
+					db.posts.insert({
+						tags: body.tags,
+						visible: !!body.visible,
```

Comment:
yeah but it's akin to doing `+x` instead of `parseInt(x)`. Functionally equivalent, but not intuitive and  hard to look at. Since there's no reason to make it compact, I vote we optimise for readability